src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -651,12 +651,14 @@ function set_diagnostic_edmf_precomputed_quantities!(Y, p, t)
         ),
     )
 
-    # TODO: This is not correct with topography or with grid stretching
+    # TODO: Currently the shear production only includes vertical gradients
     ᶠu⁰ = p.ᶠtemp_C123
     @. ᶠu⁰ = C123(ᶠinterp(Y.c.uₕ)) + C123(ᶠu³⁰)
-    ct3_unit = p.ᶜtemp_CT3
-    @. ct3_unit = CT3(Geometry.WVector(FT(1)), ᶜlg)
-    @. ᶜshear² = norm_sqr(adjoint(CA.ᶜgradᵥ(ᶠu⁰)) * ct3_unit)
+    ᶜgradᵥ_ᶠuvw⁰ = p.ᶜtemp_UVWxUVW
+    ᶜgradᵥ_ᶠuvw⁰ =
+        Geometry.project.(Ref(Geometry.UVWAxis()), ᶜgradᵥ.(UVW.(ᶠu⁰)))
+    @. ᶜshear² = norm_sqr((ᶜgradᵥ_ᶠuvw⁰ + adjoint(ᶜgradᵥ_ᶠuvw⁰)) / 2)
```

Comment:
It may be better to first define the strain rate `(ᶜgradᵥ_ᶠuvw⁰ + adjoint(ᶜgradᵥ_ᶠuvw⁰)) / 2`, since we need that for momentum diffusion, and then compute the norm of that. Instead of shear^2, a better name might be strain_norm or something like it.

Reply:
I isolated the strain rate calculation in the function compute_strain_rate! (It currently computes the strain rate at the center, we can add the one that computes it at the face when we need it.) I also renamed shear^2 to strain_rate_norm.